setup.py

```diff
@@ -62,7 +62,7 @@
         "packaging >=19.2",
         "pandas >=1.0.0, ==1.*",
         "phylo-treetime >=0.10.0, <0.12",
-        "pyfastx >=0.8.4, ==0.8.*",
+        "pyfastx >=0.9.0, ==0.9.*",
```

Comment:
Some additional changes are necessary. [Error from CI](https://github.com/nextstrain/augur/actions/runs/6778141605/job/18423099741?pr=1335#step:8:1142):

```
E   KeyError: 'SEQ_A does not exist in fasta file'
```

Reply:
Addressed with dee8a5d734468026354c26b0844662efade3a0dc...b0450991571d3536ede85f4d30d15858177bac0c

Reply:
Actually, I don't think 2e7d620429eebf2fb2c721df5be20b794ca148f3 is necessary... accessing via id as noted in the pyfastx README seems to work fine. I'll have to dig a bit deeper to see what's going wrong with the tests.

Reply:
Looks like this is a [known bug in 0.9.1](https://github.com/lmdu/pyfastx/issues/62) that has been fixed in 1.0.0. 

I don't see any backwards incompatible changes in the [pyfastx changelog](https://github.com/lmdu/pyfastx/blob/ee7303d1402d5f83db1a6ec74b5140b912ae75bb/docs/changelog.rst), so I think we can bump to 1.0.* (maybe even 2.0.* ?)

Reply:
@joverlee521 thanks for looking into this! How's a7210930b6cbd617db3c16ff3bde4889762e2f96?

Reply:
LGTM 👍 

I was slightly worried by the [pyfastx zlib issue](https://github.com/lmdu/pyfastx/issues/73) for docker-base, but looks like there are wheels available for Python 3.10 starting with version 0.9.0. 

Maybe we can revert https://github.com/nextstrain/docker-base/commit/485b3e5bff48c0bb8f5909dce4cb08990917709a once this is released.

Reply:
Good idea! Added that to the PR checklist.